File: backend/semgrep_rules/tests/python/kisa_sw_09_xml_injection.py
def unsafe(request, doc):
    query = request.GET.get("xpath")
    # ruleid: kisa.sw09.python.external-input-to-xpath
    return doc.xpath(query)

def safe(doc):
    # ok: kisa.sw09.python.external-input-to-xpath
    return doc.xpath("/users/user")

File: backend/semgrep_rules/tests/javascript/kisa_sw_09_xml_injection.js
const xpath = require("xpath");

function unsafe(req, doc) {
  const q = req.query.xpath;
  // ruleid: kisa.sw09.javascript.external-input-to-xpath
  return xpath.select(q, doc);
}
function safe(doc) {
  // ok: kisa.sw09.javascript.external-input-to-xpath
  return xpath.select("/users/user", doc);
}
